src/error/helpers/is-error-response.helper.ts

```diff
@@ -0,0 +1,13 @@
+import { ErrorResponse } from '../types/error-response.type';
+
+export const isErrorResponse = (error: string | { message?: string; statusCode?: number }): error is ErrorResponse => {
+  if (typeof error !== 'object') {
+    return false;
+  }
+
+  if (typeof error.message !== 'string' && typeof error.statusCode !== 'number') {
```

Comment:
This will return true for object which only contains message, i.e. `isErrorResponse({ message: 'Hello' })`.

Reply:
good catch 🤦